aether-kernel/aether/kernel/api/views.py

```diff
@@ -450,6 +416,57 @@ def get_entity_linked_data(entity, request, resolved, depth, start_depth=0):
         return Response(serializers.EntitySerializer(instance, context={'request': request}).data)
 
 
+class SubmissionStatsMixin(object):
+
+    search_fields = ('name',)
+    ordering_fields = ('name', 'created',)
+    ordering = ('name',)
+
+    def get_queryset(self):
+        family = self.request.query_params.get('family')
+        if family:  # filter entities by schema family
+            # Django 1: use Case+When
+            entities_count = Count(
+                Case(
+                    When(
+                        submissions__entities__projectschema__schema__family=family,
+                        then='submissions__entities__id',
+                    ),
+                ),
+                distinct=True,
+            )
+            # Django 2: filter in Count
```

Comment:
So is this a proposed alternative implementation? How can we choose between them?

Reply:
We need to migrate to Django 2 to have the second option.

https://github.com/eHealthAfrica/aether/pull/287

Reply:
Ah, I didn't realise that `1` and `2` just referred to versions! Could you just put in something like:

```suggestion
            # Django 2: filter in Count 
            # (replace code above with this block when we upgrade to Django 2)
```